virtwho/register.py

```diff
@@ -261,8 +263,195 @@ def facts_remove(self):
 
 
 class RHSMAPI:
-    def __init__(self):
-        pass
+
+    def __init__(self, host, username, password, port=22):
+        """
+        Using rhsm api to check/get/delete consumers,  attach/remove
+        subscription, and check the host-to-guest associations.
+        :param host: ip/hostname to run curl command.
+        :param username: account username of the host
+        :param password: password to access the host
+        :param port: port to access the host
+        """
+        self.ssh = SSHConnect(host=host, user=username, pwd=password, port=port)
+        self.server = config.rhsm.server
+        self.username = config.rhsm.username
+        self.password = config.rhsm.password
+        self.org = config.rhsm.default_org
+        self.api = f'https://{self.server}/subscription'
+        self.curl_base = f'curl -s -k -u {self.username}:{self.password}'
+
+    def consumers(self, host_name=None):
+        """
+        Search consumer host information.
+        :param host_name: host name, search all consumers if host_name=None.
+        :return: one consumer or all consumers to a list.
+        """
+        cmd = f'{self.curl_base} {self.api}/owners/{self.org}/consumers'
+        ret, output = self.ssh.runcmd(cmd)
+        if ret == 0 and output:
+            consumers = json.loads(output.strip())
+            if host_name:
+                for consumer in consumers:
+                    if host_name in consumer['name'].strip():
+                        return consumer
+            else:
+                return consumers
+        return None
+
+    def uuid(self, host_name):
+        """
+        Get the consumer uuid by host name
+        :param host_name: host name
+        :return: consumer uuid or None
+        """
+        consumer = self.consumers(host_name)
+        if consumer:
+            uuid = consumer['uuid'].strip()
+            logger.info(f'Succeeded to get stage consumer uuid: '
+                        f'{host_name}:{uuid}')
+            return uuid
+        raise FailException(
+            f'Failed to get stage consumer uuid for {host_name}')
+
+    def get(self, host_name):
```

Comment:
it will more make sense to rename it: info()

Reply:
Done